app/src/main/java/com/example/lastfm/MainActivity.java

```diff
@@ -52,6 +61,7 @@ public void onSuccess(List<Artist> info) {
                             requestedArtists = info;
                             showArtists(requestedArtists);
                             hideRefreshing();
+                            progressDoalog.dismiss();
```

Comment:
u must create a method because u use the same code in different places

Reply:
Created method finishProgressDialog